src/Compilers/Core/Portable/Symbols/Nullability.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+    public enum Nullability
+    {
+        NonNull,
+        CanBeNull,
+        Unknown
+    }
+}
```

Comment:
Let's discuss with Chuck. From our discussion yesterday, we thought maybe the public API should expose `IsAnnotated` (`?` or blank) plus `NonNullTypes` context as primitives rather than `IsNullable` (tri-state).

Also, you may want to take a look into the public APIs related to conversions. Maybe they also need to be extended.

Reply:
Also, FWIW, the scenarios we could think of so far, for an analyzer writer:
- should I add a `?` to this type when I generate code? (for example, IntroduceLocal)
- could I assign `null` to this type?
- could I get a `null` when reading from this type?

---
In reply to: [205584504](https://github.com/dotnet/roslyn/pull/28860#discussion_r205584504) [](ancestors = 205584504)

Reply:
Honestly, I'm not sure what `IsAnnotated` + `NonNullTypes` gets public users over a tri-state, and we'd still need this anyway. `IsAnnotated` might be true even after a null check, but the type should be non-nullable at that point.
Note that, from the name, I've determined that `IsAnnotated` refers to the declaration of the variable, not current state. If that's not what you meant, then we probably need a better name.

---
In reply to: [205585491](https://github.com/dotnet/roslyn/pull/28860#discussion_r205585491) [](ancestors = 205585491,205584504)